### PR TITLE
Use integer for plant spread and height instead of enum

### PR DESCRIPTION
**Pull request by chr-schr**
Leftover task:
 - [ ] data quality should be improved, maybe a bit can be fixed with scraper and definitely we can improve using manual overrides  #726

## Basics

- [X] I added a line to [changelog.md](/doc/changelog.md)
- [X] The PR is rebased with current master.
- [X] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [ ] The buildserver is happy. **(buildserver needs new dataset to be happy)**

## Checklist

- [X] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [X] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [X] I _fixed_ automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [X] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [X] Code is conforming to [our Architecture](/doc/architecture)
- [X] Code is conforming to [our Guidelines](/doc/guidelines)
- [X] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines

#822

### CODE REVIEW

**Comment by markus2330** [comment]
Great, looking forward that you undraft it to have a look at it! :rocket: :heart:

**Comment by markus2330** [comment]
Good that you opened the PR. Even better would be if you write a few lines in an issue (or create an issue), to clarify what exactly the scope is etc. (especially if both frontend and backend is involved). One important note, if you didn't find it already: part of the task is basically reverting what https://github.com/ElektraInitiative/PermaplanT/pull/692/files did. To fix https://github.com/ElektraInitiative/PermaplanT/pull/721/files to work with int instead of enum pls request help from frontend people as needed or we can also simply split it up in two issues. Let us talk about this on Monday.

**Comment by chr-schr** [comment]
@markus2330 
The build fails because of incompatible scraper data. Should I make a PR in the scraper repository for the required changes?
(the files need to be merged with the updated scraper)

**Comment by markus2330** [comment]
> The build fails because of incompatible scraper data. Should I make a PR in the scraper repository for the required changes?

Yes, please!

**Comment by chr-schr** [comment]
@markus2330 
I created the PR for the data change: https://github.com/ElektraInitiative/scraper-data/pull/1
These two PRs should be merged simultaneously.

While developing this PR I had to run `npm run merge:datasets` quite often and noticed that every time the datasets get merged, the german names are fetched from the Wikidata API. Those feel like two different steps for me; maybe they should be separated? IMO the german names should also be saved in a file such that changing one dataset and re-merging them does not necessitate fetching all german names again.

What do you think? Should I create a new issue to discuss this?

**Comment by Bushuo** [comment]
@chr-schr
please make sure to name the branches `feature/` with lowercase `f`.
That produced some really weird bug for me in git.

**Comment by markus2330** [comment]
@chr-schr please make sure to add "please merge" when you are done, as the CI obviously cannot succeed.

**Review by horenso — APPROVED** [review]

File: frontend/src/__test_utils__/msw_handlers/plants.ts [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+import { Page, PlantsSummaryDto } from '@/api_types/definitions';
 import { baseApiUrl } from '@/config';
 
 const allPlants: PlantsSummaryDto[] = [
   {
     id: 1,
     unique_name: 'test',
     common_name_en: ['test'],
-    spread: PlantSpread.Na,
+    spread: undefined,
```
[the remark this answers is not in the document]
Comment (chr-schr) [reply frontend/src/__test_utils__/msw_handlers/plants.ts]: Tests still pass after removing it 👍 
Thanks for the review!

**Review by horenso — APPROVED** [review]

**Comment by markus2330** [comment]
jenkins build please

**Pull request merged**